feature/VONK-7767: Add information about changes to multi-tenancy (#605)
FirelyTeam/firely-docs-firely-server: features_and_tools/multi-tenancy.rst

The basics are simple:

- On every request, the tenant is read from either a specific HTTP header or a claim in the authorization token.

> - If no tenant is specified, depending whether your operation configuration requires it, the special `shared` tenant might be used internally.

Reviewer: 'might' is not very clear wording. I suggest to be very explicit. Note also that in rst you need double backticks for inline code.
```suggestion
- If no tenant is specified, and the operation being invoked does not require a tenant, the special ``shared`` tenant is used internally.
```

- If the request creates or updates any resource, the tenant is applied to each of those resources. This is done by adding a *security label* to the ``meta.security`` element of the resource.
- For both reading and writing resources, the tenant is used to limit access to only the resources having a matching security label.

> - Upon reading, the tenant security label is (by default) removed from the resource and thus not visible from the outside.

Reviewer: Avoid parentheses in text, they make text unclear. It is either by default or not. In this case, it is.
```suggestion
- Upon reading, by default, the tenant security label is removed from the resource and thus not visible from the outside. If you want to communicate the label to the client, set ``RemoveTenantLabelFromResponse`` to ``false`` in the settings.
```

:TenantLabelSystem: The value of the ``meta.security.system`` element to use for the tenant security label. You can only choose this once.
:AllowTenantFromHeader: The tenant may be specified with an HTTP header, with the name specified in ``TenantHeader``.
:AllowTenantFromClaim: The tenant may be specified with a claim in the authorization token, with the name specified in ``TenantClaim``.
:RemoveTenantLabelFromResponse: The label will be removed from the resource by default, but this behaviour can be disabled if this option is set to ``false``.
:ResourceAccess: This structure allows to define set of conditional logic that will be applied to each request.
:ResourceAccess:Priority: This value will be used to choose most important logic entry if multiple entries would match the request.
:ResourceAccess:Tenant: This value specifies whether the tenant is required to be specified when accessing the resource. It has to be either ``Required`` or ``Shared``.
:ResourceAccess:Profiles: This value specifies a list of ``meta.profile`` values to look for in the resource. The resource should have at least one of those values.

> :ResourceAccess:ResourceTypes: This value specifies a list of resources this configuration will be applicable to. When left empty, it is assumed to be not constrained.

Reviewer: ```suggestion
:ResourceAccess:ResourceTypes: This value specifies a list of resource types this configuration will be applicable to. When left empty, it is assumed to be not constrained.
```
- 'resources' and 'resource types' are casually confused, but as FHIR pro's we should not do that. A 'resource' is an instance.
- I assume you update this based on our discussed meaning of an empty array (i.e.: nothing matches).

> :TenantHeader: The name of the HTTP header to read the tenant from. This is only evaluated if ``AllowTenantFromHeader`` is set to ``true``.
> :TenantClaim: The name of the claim in the authorization token to read the tenant from. This is only evaluated if ``AllowTenantFromClaim`` is set to ``true``.
> :TenantLabelSystem: The value of the ``meta.security.system`` element to use for the tenant security label. You can only choose this once.
> :AllowTenantFromHeader: The tenant may be specified with an HTTP header, with the name specified in ``TenantHeader``.
> :AllowTenantFromClaim: The tenant may be specified with a claim in the authorization token, with the name specified in ``TenantClaim``.
> :RemoveTenantLabelFromResponse: The label will be removed from the resource by default, but this behaviour can be disabled if this option is set to ``false``.
> :ResourceAccess: This structure allows to define set of conditional logic that will be applied to each request.
> :ResourceAccess:Priority: This value will be used to choose most important logic entry if multiple entries would match the request.
> :ResourceAccess:Tenant: This value specifies whether the tenant is required to be specified when accessing the resource. It has to be either ``Required`` or ``Shared``.
> :ResourceAccess:Profiles: This value specifies a list of ``meta.profile`` values to look for in the resource. The resource should have at least one of those values.
> :ResourceAccess:ResourceTypes: This value specifies a list of resources this configuration will be applicable to. When left empty, it is assumed to be not constrained.
> :ResourceAccess:Filter: This value specifies a filter that resources must fulfill.

Reviewer: This gets ugly to read. I suggest a nested list:
```suggestion
- TenantHeader: The name of the HTTP header to read the tenant from. This is only evaluated if ``AllowTenantFromHeader`` is set to ``true``.
- TenantClaim: The name of the claim in the authorization token to read the tenant from. This is only evaluated if ``AllowTenantFromClaim`` is set to ``true``.
- TenantLabelSystem: The value of the ``meta.security.system`` element to use for the tenant security label. You can only choose this once.
- AllowTenantFromHeader: The tenant may be specified with an HTTP header, with the name specified in ``TenantHeader``.
- AllowTenantFromClaim: The tenant may be specified with a claim in the authorization token, with the name specified in ``TenantClaim``.
- RemoveTenantLabelFromResponse: The label will be removed from the resource by default, but this behaviour can be disabled if this option is set to ``false``.
- ResourceAccess: This structure allows to define set of conditional logic that will be applied to each request.

    - Priority: This value will be used to choose most important logic entry if multiple entries would match the request.
    - Tenant: This value specifies whether the tenant is required to be specified when accessing the resource. It has to be either ``Required`` or ``Shared``.
    - Profiles: This value specifies a list of ``meta.profile`` values to look for in the resource. The resource should have at least one of those values.
    - ResourceTypes: This value specifies a list of resources this configuration will be applicable to. When left empty, it is assumed to be not constrained.
    - Filter: This value specifies a filter that resources must fulfill.
```
